File: springboot-todo/src/main/kotlin/com/example/todo/TaskController.kt
package com.example.todo

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/tasks")
class TaskController(private val taskService: TaskService) {

    @GetMapping
    fun findAll(): List<Task> = taskService.findAll()

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    fun create(@RequestBody input: TaskInput): Task = taskService.create(input)

    @DeleteMapping("/{id}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    fun delete(@PathVariable id: Int) = taskService.delete(id)
}

File: nestjs-todo/src/task.controller.ts
import { Controller, Get, Post, Delete, Body, Param } from '@nestjs/common';
import { TaskService } from './task.service';
import { Task } from './task.interface';

@Controller('api/tasks')
export class TaskController {
  constructor(private readonly taskService: TaskService) {}

  @Get()
  findAll(): Task[] {
    return this.taskService.findAll();
  }

  @Post()
  create(@Body() body: { name: string; description: string }): Task {
    return this.taskService.create(body.name, body.description);
  }

  @Delete(':id')
  delete(@Param('id') id: string): void {
    this.taskService.delete(parseInt(id, 10));
  }
}
